camayoc/tests/qcs/api/v1/scanjobs/test_run_scanjobs.py

```diff
@@ -303,3 +311,101 @@ def test_scan_task_results(scan_info):
         num_failed = task['systems_failed']
         num_scanned = task['systems_scanned']
         assert num_scanned == sys_count - num_failed
+
+
+def test_disabled_optional_products_facts():
+    """Test scan jobs from scans with disabled optional products.
+
+    :id: 6f91ea5c-32b9-11e8-b467-0ed5f89f718b
+    :description: Test that scan jobs of scans with disabled products
+        are not gathering the raw_facts defined in the roles of the
+        disabled products.
+    :steps:
+        1) Iterate over the scans and gather the disabled_optional_products
+        2) For any scan that was defined with disabled_optional_products:
+            a) Create a facts_to_ignore list composed of raw_facts for each
+                disabled role
+            b) Iterate through the inspection results fact dictionaries
+            c) Assert that no facts that should be ignored are in the
+                dictionaries
+    """
+    fuse_raw_facts = ['karaf_locate_karaf_jar', 'karaf_homes',
+                      'karaf_home_bin_fuse', 'karaf_home_system_org_jboss',
+                      'fuse_activemq_version', 'karaf_running_processes',
+                      'jboss_fuse_systemctl_unit_files',
+                      'jboss_fuse_chkconfig', 'fuse_camel_version',
+                      'fuse_cxf_version']
+    brms_raw_facts = ['internal_jboss_brms_business_central_candidates',
+                      'jboss_brms_business_central_candidates',
+                      'jboss_brms_kie_server_candidates',
+                      'business_central_candidates_eap',
+                      'kie_server_candidates_eap',
+                      'jboss_brms_locate_kie_api',
+                      'internal_jboss_brms_locate_kie_api',
+                      'jboss_brms_kie_in_business_central',
+                      'jboss_brms_manifest_mf',
+                      'business_central_candidates']
+    eap_raw_facts = ['jboss_eap_running_paths',
+                     'jboss_eap_locate_jbos_modules_jar',
+                     'eap_home_candidates', 'eap_home_ls',
+                     'eap_home_version_txt', 'eap_home_readme_txt',
+                     'eap_home_jboss_modules_manifest',
+                     'eap_home_jboss_modules_version',
+                     'eap_home_bin', 'eap_home_layers',
+                     'eap_home_layers_conf', 'jboss_eap_common_files',
+                     'jboss_eap_processes', 'jboss_eap_packages']
```

Comment:
Lets move all these up into `camayoc/constants.py` and name them things like `QCS_BRMS_RAW_FACTS` etc.

You will see in there the constants also include a doc string that gets rendered in our api docs, so you can describe what these are for.